src/sagemaker/tensorflow/estimator.py

```diff
@@ -47,6 +48,29 @@ def _cmd_exists(cmd):
             for path in os.environ["PATH"].split(os.pathsep)
         )
 
+    @staticmethod
+    def _sync_directories(from_directory, to_directory):
+        """Sync to_directory with from_directory by copying each file in
+        to_directory with new contents. Why do this? Because TensorBoard picks
+        up temp files from `aws s3 sync` and then stops reading the correct
+        tfevent files.
+
+        Args:
+            from_directory (str): The directory with updated files.
+            to_directory (str): The directory to be synced.
+        """
+        for root, dirs, files in os.walk(from_directory):
```

Comment:
I think you'll want to check to see if `to_directory` exists before this loop, then create it if it doesn't. From what I can tell, running `aws s3 sync src_dir dst_dir` will create `dst_dir` for you, but it looks like we're only attempting to create subdirectories of `to_directory` rather than the directory itself.